ci: apply `zizmor` security audit

oleonardolima opened the pull request:
fixes #1775

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Description

I used `zizmor` on all current CI workflows, it's a tool that helps detecting possible vulnerabilities in our CI jobs, see https://woodruffw.github.io/zizmor/.

It can run against most of it's audit rules, however the ones that require the GitHub API Token would require some with access to it in order to test against it. So this PR does not cover for impostor-commit, ref-confusion known-vulnerable-actions audit rules.

<!-- Describe the purpose of this PR, what's being adding and/or fixed -->

### Notes to the reviewers

<!-- In this section you can include notes directed to the reviewers, like explaining why some parts
of the PR were done in a specific way -->

### Changelog notice

- Do not persist credentials on GitHub Actions. 

<!-- Notice the release manager should include in the release tag message changelog -->
<!-- See https://keepachangelog.com/en/1.0.0/ for examples -->

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

oleonardolima committed:
fix(ci): do not persist credentials

fix(cron-update-rust): don't persist credentials

fix(audit): don't persist credentials

fix(code-coverage): don't persist credentials

fix(nightly-docs): don't persist credentials

fix(cont-integration): don't persist credentials

notmandatory reviewed (approved):
ACK 94daa75cd784cddaf3901e665b94f0a50dfed676

This just needs to be rebased and then it's ready to merge.

oleonardolima committed:
fix(cont-integration): template injection audit

- fixes the `template_injection` audit failure due to
  `matrix.rust.version` usage, use an environement var instead
  see: https://woodruffw.github.io/zizmor/audits/#template-injection

oleonardolima commented:
I did rebase it, and also pushed a new commit 30dce98 to fix the [`template-injection`](https://woodruffw.github.io/zizmor/audits/#template-injection) audit rule, as we're now manipulating the `matrix.rust.version` when running the job.

notmandatory reviewed (approved):
ACK 30dce98450fb3a919babfe512d8e8d505f3dddd8
